nunjucks/src/filters.js

```diff
@@ -266,20 +266,34 @@ function random(arr) {
 
 exports.random = random;
 
+function getSelectOrReject(testResultFn) {
+  function filter(arr, testName = 'truthy', secondArg) {
+    const context = this;
+    const test = context.env.getTest(testName);
+
+    return arr.filter(function applyToTest(item) {
+      return testResultFn(test.call(context, item, secondArg));
+    });
+  }
+
+  return filter;
+}
+
+const reject = getSelectOrReject(function examineRejectResult(testResult) {
+  return testResult === false;
+});
```

Comment:
It looks like jinja2 doesn't strictly require reject to return false from the test function, it's sufficient for it to be truthy. So it might be better to do this as:

```suggestion
const reject = getSelectOrReject((res) => !res);
const select = getSelectOrReject((res) => !!res);
```

Reply:
`testResult` here is the output of `test` function, so it is always boolean.

Reply:
I have add jsdoc to make it more clear

Reply:
Thanks!